v1/plugin/plugin.go

```diff
@@ -274,8 +280,10 @@ func StartStreamCollector(plugin StreamCollector, name string, version int, opts
 		panic(err)
 	}
 	proxy := &StreamProxy{
-		plugin:      plugin,
-		pluginProxy: *newPluginProxy(plugin),
+		plugin:             plugin,
+		pluginProxy:        *newPluginProxy(plugin),
+		maxCollectDuration: defaultMaxCollectDuration,
+		maxMetricsBuffer:   defaultMaxMetricsBuffer,
```

Comment:
**To reviewers**:
I suppose that it will be better to provide func `newStreamProxy(pl plugin)`, but something similar should be done for others existing proxy (e.g. [collectorProxy](https://github.com/intelsdi-x/snap-plugin-lib-go/blob/master/v1/plugin/plugin.go#L227) and [publisherProxy](https://github.com/intelsdi-x/snap-plugin-lib-go/blob/master/v1/plugin/plugin.go#L259), etc.), so I decide not to do that in this PR to keep this PR as clear as possible.